lib/GHCup/Platform.hs

```diff
@@ -119,11 +119,17 @@ getPlatform = do
         either (const Nothing) Just . versioning . decUTF8Safe'
           <$> getFreeBSDVersion
       pure $ PlatformResult { _platform = FreeBSD, _distroVersion = ver }
+    "openbsd" -> do
+      ver <-
+        either (const Nothing) Just . versioning . decUTF8Safe'
+          <$> getOpenBSDVersion
+      pure $ PlatformResult { _platform = OpenBSD, _distroVersion = ver }
     "mingw32" -> pure PlatformResult { _platform = Windows, _distroVersion = Nothing }
     what -> throwE $ NoCompatiblePlatform what
   lift $ logDebug $ "Identified Platform as: " <> T.pack (prettyShow pfr)
   pure pfr
  where
+  getOpenBSDVersion = lift $ fmap _stdOut $ executeOut "uname" ["-r"] Nothing
```

Comment:
Are you sure `uname` is the *OpenBSD* version and not just the kernel?

What are the outputs? Are the purely numeric? Otherwise we will have a hard time matching on them.

Reply:
I don't think the BSDs have a kernel version distinct from the OS version, since they're all developed by the same people under a single umbrella; a release _is_ the kernel, plus base packages (userland), plus ports (package repository for that version).

I developed these changes on my real OpenBSD/amd64 machine and tested the `uname -r` command after reading the man page at the time.

OpenBSD's man page for `uname` says:

>```
>     -r      Print the operating system release.  On OpenBSD, the format is
>             digit.digit.
>```

(and digit.digit is underlined, but (GitHub's) Markdown doesn't seem to support that, nor setting monospace font quote without a code block which makes things literal anyway.)

However, I just realised the man page doesn't specify, and I hadn't considered, what would happen on OpenBSD-snapshot or -current (which are non-stable tracks that some OpenBSD users do prefer)? I'm gonna check on a VM now.

Reply:
Comments on [this Reddit thread](https://old.reddit.com/r/openbsd/comments/1b604uv/snapshot_vs_release) basically confirm that there's no such thing as a -snapshot track, so that was my bad; snapshots are just compiled releases of the -current track.